model-server/src/main/kotlin/org/modelix/model/server/handlers/DeprecatedLightModelServer.kt

```diff
@@ -144,95 +142,97 @@ class DeprecatedLightModelServer(val client: LocalModelClient) {
-                        sendDelta(newVersion)
+                val listener = object : IKeyListener {
+                    override fun changed(key: String, value: String?) {
+                        if (value == null) return
+                        launch {
+                            repositoriesManager.runWithRepository(repositoryId) {
+                                val newVersion = CLVersion.loadFromHash(value, client.storeCache)
+                                sendDelta(newVersion)
+                            }
+                        }
                     }
                 }
-            }
 
-            client.listen(repositoryId.getBranchKey(), listener)
-            try {
-                sendDelta(getCurrentVersion(repositoryId))
-                for (frame in incoming) {
-                    when (frame) {
-                        is Frame.Text -> {
-                            val updateData = JSONArray(frame.readText())
-                            val mergedVersion = applyUpdate(lastVersion!!, updateData, repositoryId, userId)
-                            sendDelta(mergedVersion)
+                client.listen(repositoryId.getBranchKey(), listener)
+                try {
+                    sendDelta(getCurrentVersion(repositoryId))
+                    for (frame in incoming) {
+                        when (frame) {
+                            is Frame.Text -> {
+                                val updateData = JSONArray(frame.readText())
+                                val mergedVersion = applyUpdate(lastVersion!!, updateData, repositoryId, userId)
+                                sendDelta(mergedVersion)
+                            }
+                            else -> {}
                         }
-                        else -> {}
                     }
+                } finally {
+                    client.removeListener(repositoryId.getBranchKey(), listener)
                 }
-            } finally {
-                client.removeListener(repositoryId.getBranchKey(), listener)
             }
         }
         post<Paths.jsonRepositoryIdInitPost> {
             // TODO error if it already exists
             val repositoryId = RepositoryId(call.parameters["repositoryId"]!!)
-            val newTree = CLTree.builder(getStore()).repositoryId(repositoryId).build()
-            val userId = call.getUserName()
-            val newVersion = CLVersion.createRegularVersion(
-                client.idGenerator.generate(),
-                Date().toString(),
-                userId,
-                newTree,
-                null,
-                emptyArray(),
-            )
-            client.asyncStore!!.put(repositoryId.getBranchKey(), newVersion.hash)
-            respondVersion(newVersion)
+            repositoriesManager.runWithRepository(repositoryId) {
+                val userId = call.getUserName()
+                val newVersion = repositoriesManager.createRepository(repositoryId, userId)
```

Comment:
I don't really care. This whole API is deprecated and unused. Just left it there because the unit tests indirectly test all the other components. This is still useful at least for this migration.